types/google-protobuf/index.d.ts

```diff
@@ -195,25 +195,25 @@ export class Map<K, V> {
   serializeBinary(
     fieldNumber: number,
     writer: BinaryWriter,
-    keyWriterFn: (writer: BinaryWriter, field: number, key: K) => void,
-    valueWriterFn: ((writer: BinaryWriter, field: number, value: V) => void)
-                   | ((writer: BinaryWriter, field: number, value: V, opt_valueWriterCallback: (value: V, writer: BinaryWriter) => void) => void),
-    opt_valueWriterCallback?: (value: V, writer: BinaryWriter) => void
+    keyWriterFn: (field: number, key: K) => void,
```

Comment:
Dropping the `BinaryWriter` seems like a pretty big change.  Can you provide some more details on why that was appropriate?

Reply:
Sorry, I did not drop it from `valueWriterFn`:

```typescript
valueWriterFn: (field: number, value: V, writeCallback?: BinaryWriteCallback) => void
```
I just combined the two function signatures to only one where the writeCallback argument is optional. In the end, the writeCallback is used by `google-protobuf` when the internal `valueCtor_` is set.

Reply:
Thanks for clarifying!  As you say, it's good to write tests. :smile: